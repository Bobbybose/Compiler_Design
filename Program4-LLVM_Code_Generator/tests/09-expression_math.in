int a = 2;
float b = 3.1;
char c = 'a';
int d;
int e;
d = a + c;
e = d - b;
e = e * a;
e = e / a;
return e;
// Should return 95