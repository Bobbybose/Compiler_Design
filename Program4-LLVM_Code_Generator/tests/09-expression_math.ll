target datalayout = "e-m:e-p270:32:32-p271:32:32-p272:64:64-i64:64-f80:128-n8:16:32:64-S128"
target triple = "x86_64-pc-linux-gnu"

define dso_local i32 @main() {
  %t1 = alloca i32
  store i32 2, i32* %t1
  %t2 = alloca float
  store float 0x4008CCCCC0000000, float* %t2
  %t3 = alloca i8
  store i8 97, i8* %t3
  %t4 = alloca i32
  %t5 = alloca i32
  %t6 = load i32, i32 *%t1
  %t7 = load i8, i8 *%t3
  %t8 = sext i8 %t7 to i32
  %t9 = add i32 %t6, %t8
  store i32 %t9, i32* %t4
  %t10 = load i32, i32 *%t4
  %t11 = load float, float *%t2
  %t12 = sitofp i32 %t10 to float
  %t13 = fsub float %t12, %t11
  %t14 = fptosi float %t13 to i32
  store i32 %t14, i32* %t5
  %t15 = load i32, i32 *%t5
  %t16 = load i32, i32 *%t1
  %t17 = mul i32 %t15, %t16
  store i32 %t17, i32* %t5
  %t18 = load i32, i32 *%t5
  %t19 = load i32, i32 *%t1
  %t20 = sdiv i32 %t18, %t19
  store i32 %t20, i32* %t5
  %t21 = load i32, i32 *%t5
  ret i32 %t21
}
