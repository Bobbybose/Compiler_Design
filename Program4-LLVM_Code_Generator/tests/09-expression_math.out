target datalayout = "e-m:e-p270:32:32-p271:32:32-p272:64:64-i64:64-f80:128-n8:16:32:64-S128"
target triple = "x86_64-pc-linux-gnu"

define dso_local i32 @main() {
  %t1 = alloca i32
  store i32 2, i32* %t1
  %t2 = alloca float
  store float 0x4008CCCCC0000000, float* %t2
  %t3 = alloca i8
  store i8 97, i8* %t3
  %t4 = alloca i32
  %t5 = alloca i32
  %t6 = load i32, i32 *%t1
  %t7 = load float, float *%t2
  %t8 = sitofp i32 %t6 to float
  %t9 = fadd float %t8, %t7
  %t10 = fptosi float %t9 to i32
  store i32 %t10, i32* %t4
  %t11 = load i32, i32 *%t4
  %t12 = load i8, i8 *%t3
  %t13 = sext i8 %t12 to i32
  %t14 = sub i32 %t11, %t13
  store i32 %t14, i32* %t5
  %t15 = load i32, i32 *%t4
  %t16 = load i32, i32 *%t1
  %t17 = mul i32 %t15, %t16
  store i32 %t17, i32* %t5
  %t18 = load i32, i32 *%t4
  %t19 = load i32, i32 *%t1
  %t20 = sdiv i32 %t18, %t19
  store i32 %t20, i32* %t5
  ret i32 0
}
